### PR TITLE
Release 0.12.0 preparation

### PR DESCRIPTION
**Pull request by tuturto**
Working towards releasing 0.12.0. This PR updates NEWS to contain list of changes since last release and updates version.py to new version. #1135 isn't merged yet, since I'm a bit unsure about it.

All comments and corrections welcome.

Relates to #1126

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,67 @@
+Changes from 0.11.0
+
+This release brings some quite significant changes on the language and as a
+result very large portion of previously written Hy programs will require
```
Comment (olasd) [inline NEWS]: s/portion/portions/

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,67 @@
...
+changes. At the same time, documentation and error messages were improved,
+hopefully making the language easier to use.
+
+   [ Language Changes ]
+   * New syntax for let, with and defclass
+   * defmacro will raise an error on &kwonly, &kwargs and &key arguments
+   * Keyword arguments to functions are required to be strings
```
Comment (olasd) [inline NEWS]: I guess that's the keyword argument labels.

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+   * slice replaced with cut
```
Comment (olasd) [inline NEWS]: ... to stop overloading the python built-in

File: NEWS [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+   * removed throw, catch, progn, defun, lisp-if, lisp-if-not and filterfalse
+   * global now takes multiple arguments
+   * Nonlocal keyword (Python 3 only)
+   * Set literals
```
Comment (olasd) [inline NEWS]: add an example ?

File: NEWS [hunk 5]
```diff
@@ -1,3 +1,67 @@
...
+   * Keyword-only arguments
```
Comment (olasd) [inline NEWS]: Python3 only

File: NEWS [hunk 6]
```diff
@@ -1,3 +1,67 @@
...
+   * Setv can assign multiple variables at once
+   * Empty form allowed for setv, del and cond
+   * One-argument division and rationals
+   * partition form for chunking collection to n-sized tuples
+   * defn-alias and demacro-alias moved into hy.contrib.alias
+   * Implicit nil in --init-- method
```
Comment (olasd) [inline NEWS]: "Stop returning the last form in --init--" ?

File: NEWS [hunk 7]
```diff
@@ -1,3 +1,67 @@
...
+   * for and cond can take multi-expression body
```
Comment (olasd) [inline NEWS]: "a" multi-expression body, I guess.

File: NEWS [hunk 8]
```diff
@@ -1,3 +1,67 @@
...
+   * Hex and octel support for integer literals
```
Comment (olasd) [inline NEWS]: s/octel/octal/

File: NEWS [hunk 9]
```diff
@@ -1,3 +1,67 @@
...
+   * Apply now mangles strings and keywords according to Hy mangling rules
+   * One arity comparison operators
```
Comment (olasd) [inline NEWS]: Might want to bundle that up with the unary division.

File: NEWS [hunk 10]
```diff
@@ -1,3 +1,67 @@
...
+   * Better error messages with hy2py and hyc
+   * Cmdline error to string conversion.
+   * In python 3.3+, generator functions always return a value
+   * &rest can be used after &optional
+
+   [ Misc. Improvements ]
+   * Allow specification of global table and module name for (eval ...)
+   * General documentation improvements
+   * Contrib.walk: Coerce non-list iterables into list form
+   * *args and **kwargs no longer have own args in Python 3.5
```
Comment (olasd) [inline NEWS]: ???

File: NEWS [hunk 11]
```diff
@@ -1,3 +1,67 @@
...
+   * Flow macros (case and switch)
+   * ap-pipe and ap-compose macros
+   * #@ reader macro for with-decorator
+   * Type check `eval` parameters
+   * `and` and `or` short-circuit
+   * `and` and `or` accept zero or more arguments
```
Comment (olasd) [inline NEWS]: redundant with language changes

Comment (tuturto) [inline NEWS]: This is e54d4be and the one in language changes is 0a942a0 (If I understood your intent correctly "One arity comparison operators"). I would keep them separate I guess.

File: NEWS (hunk 10 above)
Comment (zackmdavis) [inline NEWS]: a implementation detail regarding changes to the Python AST; if mentioning in the release notes at all, probably fold under "Python 3.5 compatibility"

Comment (zackmdavis) [inline NEWS]: well, Python 3.5 compatibility itself is interesting; Hy 0.11 will misbehave under Py3.5 (see [the pull request comment](https://github.com/hylang/hy/pull/800))

Comment (tuturto) [inline NEWS]: Right, I should learn to read whole sentences.

**Comment by Kodiologist** [comment]
@tuturto Do you have any concerns about #1135? version.py will indeed be there so long as you run setup.py before uploading to PyPI.

**Comment by tuturto** [comment]
@Kodiologist nothing anymore, now that I had chance to go through the code with some thought. Looks good to me.

**Comment by tuturto** [comment]
uh, why did this closed automatically? I'll check what's going on.

**Comment by tuturto** [comment]
Apparently I don't know anymore how to rebase properly :D Will clean up the mess and push changes again to my topic branch.

**Pull request closed without merging**